skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/user/User.java

```diff
@@ -385,4 +388,108 @@ public boolean equals(Object o) {
 	public int hashCode() {
 		return Objects.hash(id, email, mobileNumber);
 	}
+
+	////////////////////////////////////////////////////////////////////////////////////
+	// FOLLOWABLE Methods
+	////////////////////////////////////////////////////////////////////////////////////
+
+	public Set<FollowStateTracker> getFollowableSet() { return this.followableSet; }
+	public void setFollowableSet(Set<FollowStateTracker> followableSet) { this.followableSet = followableSet; }
+
+	@Override
+	public FollowableStatus getFollowableStatus() { return this.followableStatus; }
+	@Override
+	public void setFollowableStatus(FollowableStatus status) { this.followableStatus = status; }
+
+	@Override
+	public FollowableNotification getFollowableNotifiable() { return this.followableNotifiable; }
+	@Override
+	public void setFollowableNotifiable(FollowableNotification followableNotifiable) { this.followableNotifiable = followableNotifiable; }
+
+	@Override
+	public Set<User> getFollowers() {
+		return this.followableSet.stream()
+			.map(item -> item.getFollower())
+			.collect(Collectors.toSet());
+	}
+
+	@Override
+	public void banFollower(User follower) {
+		this.followableSet.forEach(item -> {
+					if ( follower.getId() == item.getFollower().getId()) {
+						item.setFollowerStatus(FollowerStatus.banned);
+					}});
+	}
+
+	@Override
+	public void notify(Set<Notification> notifications) {
+		this.followableSet.forEach(item -> { item.pushNotifications(notifications); });
+	}
+
+	////////////////////////////////////////////////////////////////////////////////////
+	// FOLLOWER Methods
+	////////////////////////////////////////////////////////////////////////////////////
+
+	public Set<FollowStateTracker> getFollowerSet() { return this.followerSet; }
+	public void setFollowerSet(Set<FollowStateTracker> followerSet) { this.followerSet = followerSet; }
+
+	@Override
+	public void follow(User followable) {
+		this.followerSet.add(new FollowStateTracker(followable, this));
+	}
+
+	@Override
+	public void unfollow(User followable) {
+		Optional<FollowStateTracker> fst = this.followerSet.stream()
+				.map(item -> {
```

Comment:
use `filter()` here, it will allow you to avoid `null` values in your stream (I do not know what happens with such values).

Reply:
altight ! implemented in next commit